gnomad-google-project/project.tf

```diff
@@ -71,3 +74,31 @@ resource "google_project_iam_member" "primary_iap_tunnel_access" {
   role    = "roles/iap.tunnelResourceAccessor"
   member  = var.primary_user_principal
 }
+
+resource "google_project_iam_member" "primary_user_artifact_reg_admin" {
+  project = google_project.current_project.project_id
+  role    = "roles/artifactregistry.admin"
+  member  = var.primary_user_principal
+}
+
+# If one is provided, grant storage read/write and artifact registry read access to a hail batch service account
+resource "google_project_iam_member" "hail_batch_object_admin" {
+  count   = length(var.hail_batch_service_account) > 0 ? 1 : 0
```

Comment:
this is saying that the count is 1 if the length of `var.hail_batch_service_account` is non-zero, right? do the rest of the fields in this resource module (e.g., `role`) not assign if count == 0?

Reply:
Yup -- Terraform doesn't have an `if` statement. So when when we have something that we only want to create under certain circumstances, we abuse the `count` parameter with a conditional expression. So, by default, that variable is a empty string (length 0), so the count gets set to 0 and the resource is not created. If the length of that variable string is more than 0, it sets the count to 1 and creates the resource.

Reply:
ohh this makes sense, thanks!